examples/cfd/01_convection_revisited.ipynb

```diff
@@ -391,10 +391,76 @@
    },
    "outputs": [
     {
-     "output_type": "stream",
      "name": "stdout",
+     "output_type": "stream",
      "text": [
-      "#define _POSIX_C_SOURCE 200809L\n#define START_TIMER(S) struct timeval start_ ## S , end_ ## S ; gettimeofday(&start_ ## S , NULL);\n#define STOP_TIMER(S,T) gettimeofday(&end_ ## S, NULL); T->S += (double)(end_ ## S .tv_sec-start_ ## S.tv_sec)+(double)(end_ ## S .tv_usec-start_ ## S .tv_usec)/1000000;\n\n#include \"stdlib.h\"\n#include \"math.h\"\n#include \"sys/time.h\"\n\nstruct dataobj\n{\n  void *restrict data;\n  int * size;\n  int * npsize;\n  int * dsize;\n  int * hsize;\n  int * hofs;\n  int * oofs;\n} ;\n\nstruct profiler\n{\n  double section0;\n  double section1;\n  double section2;\n} ;\n\n\nint Kernel(const float dt, const float h_x, const float h_y, struct dataobj *restrict u_vec, const int i0x_ltkn, const int i0x_rtkn, const int i0y_ltkn, const int i0y_rtkn, const int time_M, const int time_m, const int x_M, const int x_m, const int y_M, const int y_m, struct profiler * timers)\n{\n  float (*restrict u)[u_vec->size[1]][u_vec->size[2]] __attribute__ ((aligned (64))) = (float (*)[u_vec->size[1]][u_vec->size[2]]) u_vec->data;\n\n  for (int time = time_m, t0 = (time)%(2), t1 = (time + 1)%(2); time <= time_M; time += 1, t0 = (time)%(2), t1 = (time + 1)%(2))\n  {\n    /* Begin section0 */\n    START_TIMER(section0)\n    for (int i0x = i0x_ltkn + x_m; i0x <= -i0x_rtkn + x_M; i0x += 1)\n    {\n      for (int i0y = i0y_ltkn + y_m; i0y <= -i0y_rtkn + y_M; i0y += 1)\n      {\n        u[t1][i0x + 1][i0y + 1] = dt*(-(-u[t0][i0x][i0y + 1]/h_x + u[t0][i0x + 1][i0y + 1]/h_x) - (-u[t0][i0x + 1][i0y]/h_y + u[t0][i0x + 1][i0y + 1]/h_y) + u[t0][i0x + 1][i0y + 1]/dt);\n      }\n    }\n    STOP_TIMER(section0,timers)\n    /* End section0 */\n    /* Begin section1 */\n    START_TIMER(section1)\n    for (int y = y_m; y <= y_M; y += 1)\n    {\n      u[t1][1][y + 1] = 1.00000000000000F;\n      u[t1][81][y + 1] = 1.00000000000000F;\n    }\n    STOP_TIMER(section1,timers)\n    /* End section1 */\n    /* Begin section2 */\n    START_TIMER(section2)\n    for (int x = x_m; x <= x_M; x += 1)\n    {\n      u[t1][x + 1][81] = 1.00000000000000F;\n      u[t1][x + 1][1] = 1.00000000000000F;\n    }\n    STOP_TIMER(section2,timers)\n    /* End section2 */\n  }\n\n  return 0;\n}\n\n"
+      "#define _POSIX_C_SOURCE 200809L\n",
```

Comment:
At this point I am not sure if I do sth wrong, the code is printed in multiple lines instead of only one. HOw would you take care of this?

Reply:
I think the change to the new format/version is way more readable.

Reply:
OK